shared/src/main/scala/tastyquery/ast/Types.scala

```diff
@@ -62,6 +62,20 @@ object Types {
 
   abstract class Type {
 
+    final def paramInfos: List[Type] = this match
+      case tpe: MethodType => tpe.paramTypes
+      case tpe             => Nil
+
+    final def tparamBounds: List[TypeBounds] = this match
+      case tpe: PolyType => tpe.paramTypeBounds
+      case tpe           => Nil
+
+    final def returnType: Type = this match
+      case tpe: MethodType => tpe.resultType
+      case tpe: PolyType   => tpe.resultType
+      case tpe: ExprType   => tpe.resType
+      case tpe             => tpe
```

Comment:
this is just the convention of both scala 2 and scala 3 compilers, I think its because most of the time it is used as a "strip whatever parameters exist, or not, I just need the result"

Reply:
Meh. I don't think that's a great convention for general usage. Again, in what circumstance would you *ever* want to "strip whatever parameters exist, I just need the result"? I can see a case made for stripping an `ExprType` or otherwise keep the type. But skipping parameters or just not skipping them? That doesn't seem reasonable in any situation.

Reply:
fixed